client/src/services/Network.ts

```diff
@@ -25,14 +34,64 @@ export default class Network {
         ? `wss://sky-office.herokuapp.com`
         : `${protocol}//${window.location.hostname}:2567`
     this.client = new Client(endpoint)
+    this.joinLobbyRoom().then(() => {
+      store.dispatch(setLobbyJoined(true))
+    })
 
     phaserEvents.on(Event.MY_PLAYER_NAME_CHANGE, this.updatePlayerName, this)
     phaserEvents.on(Event.MY_PLAYER_TEXTURE_CHANGE, this.updatePlayer, this)
     phaserEvents.on(Event.PLAYER_DISCONNECTED, this.playerStreamDisconnect, this)
   }
 
-  async join() {
-    this.room = await this.client.joinOrCreate('skyoffice')
+  /**
+   * method to join Colyseus' built-in LobbyRoom, which automatically notifies
+   * connected clients whenever rooms with "realtime listing" have updates
+   */
+  async joinLobbyRoom() {
+    this.lobby = await this.client.joinOrCreate(RoomType.LOBBY)
+
+    this.lobby.onMessage('rooms', (rooms) => {
+      store.dispatch(setAvailableRooms(rooms))
+    })
+
+    this.lobby.onMessage('+', ([roomId, room]) => {
+      store.dispatch(addAvailableRooms({ roomId, room }))
+    })
+
+    this.lobby.onMessage('-', (roomId) => {
+      store.dispatch(removeAvailableRooms(roomId))
+    })
+  }
+
+  // method to join the public lobby
+  async joinOrCreatePublic() {
+    this.room = await this.client.joinOrCreate(RoomType.PUBLIC)
+    this.initialize()
+  }
+
+  // method to join a custom room
+  async joinCustomById(roomId: string, password: string | null) {
+    this.room = await this.client.joinById(roomId, { password })
+    this.initialize()
+  }
+
+  // method to create a custom room
+  async createCustom(roomData: IRoomData) {
+    const { name, description, password, autoDispose } = roomData
+    this.room = await this.client.create(RoomType.CUSTOM, {
+      name,
+      description,
+      password,
```

Comment:
note that you're sending password as plain text here, so this is only safe because we're using HTTPS in prod mode (and `wss` for websockets).
nothing to change here, just FYI if this is ever deploy to HTTP (not HTTPS) anyone can sniff the network packets and see all the passwords sent between client and server

Reply:
Gotcha